packages/babel-types/src/builders/generated/index.ts

```diff
@@ -2,22 +2,31 @@
  * This file is auto-generated! Do not modify it directly.
  * To re-generate run 'make build'
  */
-import builder from "../builder";
+import validateNode from "../validateNode";
 import type * as t from "../..";
-
-/* eslint-disable @typescript-eslint/no-unused-vars */
-
 export function arrayExpression(
-  elements?: Array<null | t.Expression | t.SpreadElement>,
-): t.ArrayExpression {
-  return builder.apply("ArrayExpression", arguments);
+  elements: Array<null | t.Expression | t.SpreadElement> = [],
+) {
+  const node: t.ArrayExpression = {
+    type: "ArrayExpression",
+    elements,
+  };
+  validateNode(node);
+  return node;
```

Comment:
To generate slightly smaller code, we could modify `validateNode` like this:
```ts
function validateNode<N extends t.Node>(node: N): N {
  // todo: because keys not in BUILDER_KEYS are not validated - this actually allows invalid nodes in some cases
  const keys = BUILDER_KEYS[node.type];
  for (const key of keys) {
    validate(node, key, node[key]);
  }
  return node;
}
```

and then do
```ts
export function arrayExpression(
     elements: Array<null | t.Expression | t.SpreadElement> = [],
) {
  return validateNode<t.ArrayExpression>({
    type: "ArrayExpression",
    elements,
  });
}
```